lib/matplotlib/pyplot.py

```diff
@@ -1204,8 +1201,8 @@ def close(fig: None | int | str | Figure | Literal["all"] = None) -> None:
     elif isinstance(fig, Figure):
         _pylab_helpers.Gcf.destroy_fig(fig)
     else:
-        raise TypeError("close() argument must be a Figure, an int, a string, "
-                        "or None, not %s" % type(fig))
+        _api.check_isinstance(
+            (Figure, int, str, None), fig=fig)  # type: ignore[unreachable]
```

Comment:
The comment should go on the first line, I think.
```suggestion
        _api.check_isinstance(  # type: ignore[unreachable]
            (Figure, int, str, None), fig=fig)
```